Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/CytoscapeJsonCreator.java

```diff
@@ -89,4 +124,108 @@ public String getJson(Map<String, Node> nodes, List<Link> links) {
+            dataObj.setConceptName(node.getLabel());
+            dataObj.setConceptUri(node.getConceptId());
+            dataObj.setConceptId(node.getConceptIdShort());
+            if(node instanceof PredicateNode){
+                dataObj.setGroup("0");
+            }
+            else{
+                dataObj.setGroup("1");
+            }
+            dataObj.setSourceReference(node.getSourceReference());
+            dataObj.setStatementIds(node.getStatementIds());
+            cytoscapeNodeList.add(createCytoscapeNodeObject(dataObj));
+        }
+        return cytoscapeNodeList;
+    }
+    
+    @Override
+    public List<CytoscapeLinkObject> getLinks(List<Link> links){
+
+        List<CytoscapeLinkObject> cytoscapeLinkList = new ArrayList<CytoscapeLinkObject>();
+        int i = 0;
+        for(Link link : links){
+            CytoscapeLinkDataObject dataObj = new CytoscapeLinkDataObject();
+            dataObj.setId(""+i);
+            dataObj.setSource(link.getSubject().getId());
+            dataObj.setTarget(link.getObject().getId());
+            dataObj.setLabel(link.getLabel());
+            dataObj.setSourceReference(link.getSourceReference());
+            dataObj.setStatementIds(new ArrayList<String>());
+            dataObj.getStatementIds().add(link.getStatementId());
+            cytoscapeLinkList.add(createCytoscapeLinkObject(dataObj));
+            i++;
+        }
+        return cytoscapeLinkList;
+    }
+    
+    private CytoscapeNodeObject createCytoscapeNodeObject(CytoscapeNodeDataObject dataObj){
+        CytoscapeNodeObject obj = new CytoscapeNodeObject();
+        obj.setData(dataObj);
+        return obj;
+    }
+    
+    private CytoscapeLinkObject createCytoscapeLinkObject(CytoscapeLinkDataObject dataObj){
+        CytoscapeLinkObject obj = new CytoscapeLinkObject();
+        obj.setData(dataObj);
+        return obj;
+    }
+    
+    @Override
+    public String submitTransformationRequest(String conceptId, IProject project){
+        String conceptIdToken = conceptId.substring(conceptId.lastIndexOf('/') + 1);
+        if(!cache.isKeyInCache(conceptIdToken)){
+              Callable<PublicSearchObject> callable = () -> {
+
+                  PublicSearchObject publicSearchObject = new PublicSearchObject();
+                  ITransformedNetwork transformedNetwork = transformationManager
+                    .getSearchTransformedNetwork(project.getProjectId(), conceptId, INetworkStatus.APPROVED);
+                  if (transformedNetwork != null && transformedNetwork.getNodes().size() > 0) {
+                      publicSearchObject.setNodes(getNodes(new ArrayList<Node>(transformedNetwork.getNodes().values())));
+                      publicSearchObject.setLinks(getLinks(transformedNetwork.getLinks()));
+                  }
+                  else{
+                      publicSearchObject.setNetworkEmpty(true);
+                  }
+                  return publicSearchObject;
+              };
```

Comment:
this should be in its own class. there should be one class per task (token management, performing a long running task, creating json, etc) -> SRP (single responsibility principle) -> a class should have only one reason to change.